implementations/rust/ockam/ockam/src/delay.rs

```diff
@@ -0,0 +1,77 @@
+use crate::{block_future, spawn, Address, Context, Message, Result, Route};
+use std::time::Duration;
+
+/// Send a delayed event to a worker
+pub struct DelayedEvent<M: Message> {
+    route: Route,
+    ctx: Context,
+    d: Duration,
+    msg: M,
+}
+
+impl<M: Message> DelayedEvent<M> {
+    /// Create a new 100ms delayed message event
+    pub fn new(ctx: &Context, route: Route, msg: M) -> Result<Self> {
+        let ctx = block_future(&ctx.runtime(), async {
+            ctx.new_context(Address::random(0)).await
+        })?;
+
+        debug!("Creating a delayed event with address '{}'", ctx.address());
+
+        Ok(Self {
+            route,
+            ctx,
+            d: Duration::from_millis(100),
+            msg,
+        })
+    }
+
+    /// Adjust the delay time with a [`Duration`](std::time::Duration)
+    pub fn duration(self, d: Duration) -> Self {
```

Comment:
Let's stick to rust idiomatic naming and start setters with set_

Reply:
Or call it something like "with_duration", because "duration" is what getter would be called.